src/libs/collect.c

```diff
@@ -1530,95 +1579,61 @@ static void menuitem_and(GtkMenuItem *menuitem, dt_lib_collect_rule_t *d)
   {
     char confname[200] = { 0 };
     snprintf(confname, sizeof(confname), "plugins/lighttable/collect/mode%1d", active);
-    dt_conf_set_int(confname, DT_LIB_COLLECT_MODE_AND);
-    snprintf(confname, sizeof(confname), "plugins/lighttable/collect/string%1d", active);
-    dt_conf_set_string(confname, "");
-    dt_conf_set_int("plugins/lighttable/collect/num_rules", active + 1);
-    dt_lib_collect_t *c = get_collect(d);
-    c->active_rule = active;
-  }
-  dt_collection_update_query(darktable.collection);
-}
-
-static void menuitem_or(GtkMenuItem *menuitem, dt_lib_collect_rule_t *d)
-{
-  // add next row with or operator
-  const int _a = dt_conf_get_int("plugins/lighttable/collect/num_rules");
-  const int active = CLAMP(_a, 1, MAX_RULES);
-  if(active < MAX_RULES)
-  {
-    char confname[200] = { 0 };
-    snprintf(confname, sizeof(confname), "plugins/lighttable/collect/mode%1d", active);
-    dt_conf_set_int(confname, DT_LIB_COLLECT_MODE_OR);
+    dt_lib_collect_mode_t mode = 0;
+    const gchar *label = gtk_menu_item_get_label (menuitem);
+    if(strcmp(label, _("narrow down search")) == 0)
```

Comment:
I don't like comparing translated strings. Instead you should store the mode in the menuitem, similar to https://github.com/darktable-org/darktable/blob/master/src/gui/presets.c#L723 and then retrieve it here (https://github.com/darktable-org/darktable/blob/master/src/gui/presets.c#L582) – just with int instead of char.